### PR TITLE
Fix #159. Add Functions list into Function App module in Azure Explorer

### LINKED ISSUE #159: Functions: list functions for a particular function app in Azure explorer
We need to list all available functions inside a particular function app. There is no API in SDK that we could use. We should rather make a REST call to get this information.

Please see comment from @maartenba in https://github.com/JetBrains/azure-tools-for-intellij/pull/158

Implementation:
AzureExplorerFunctions

Notes: Currently there are no actions to disable/enable azure function because there is no strict and easy way to define them. Please see:
- https://github.com/Azure/azure-functions-ux/issues/949
- https://github.com/Azure/azure-functions-host/issues/2623

### PR DESCRIPTION
**Pull request by sdubov**
- Add logic to get list of available functions for a function app in Azure Explorer
- Add placeholders for action to enable/disable a function (currently, Azure API does not provide a strict function state - see comment in code)
- Add dependency on jackson-modile-kotlin in azuretools-core module

### CODE REVIEW

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/functionapp/functions/Function.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+interface Function :
+        HasName,
+        GroupableResource<AppServiceManager, SiteInner>,
+        Refreshable<Function>,
+        HasParent<FunctionApp> {
+
+    // This is currently the placeholder until issue is fixed - https://github.com/Azure/azure-functions-host/issues/2623
+    // There is no strict way to define current azure function status
+    fun isEnabled(): Boolean
```
Comment (maartenba) [inline Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/functionapp/functions/Function.kt:41]: Ugh. I thought for V2 functions (we don't really support V1) this is done via appsettings: https://docs.microsoft.com/en-us/azure/azure-functions/disable-function#functions-2x---all-languages

But then there is a note saying the V1 version is also supported in V2 (https://docs.microsoft.com/en-us/azure/azure-functions/disable-function#functions-1x---c-class-libraries).

Maybe we can park enable/disable for now.

Comment (maartenba) [reply Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/functionapp/functions/Function.kt:41]: Created #182, let's tackle this later.

**Review by maartenba — APPROVED** [review]
Looks good to me, ready for merge.

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/functionapp/functions/Function.kt (hunk 1 above)
Comment (sdubov) [reply Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/functionapp/functions/Function.kt:41]: Agree. Another note on the same page saying - "Also, note that the portal may indicate the function is disabled when it isn't." :)

**Pull request merged**